build-images/github-webhook-monitor/cmd/main.go

```diff
@@ -210,26 +217,30 @@ func buildHookRequest(id string) (*http.Request, error) {
 	if err != nil {
 		log.Printf("buildHookRequest - Failed retrieving webhook request body. %v\n", err)
 	} else {
+		if resp.StatusCode == http.StatusNotFound {
+			log.Printf("buildHookRequest - unable to unmarshal into json request struct because event %s has response code of 404", id)
+			err = fmt.Errorf(EVENT_ID_NOT_FOUND_ERROR_MSG)
+		} else {
+			err = json.Unmarshal(b, &request)
+			if err == nil {
 
-		err = json.Unmarshal(b, &request)
-		if err == nil {
+				if eventType, ok := triggerMap.Events[request.Event]; ok {
+					url := eventType.EventListener
+					payload, _ := json.Marshal(request.Request.Payload)
+					webhookRequest, err := http.NewRequest("POST", url, bytes.NewReader(payload))
+					if err != nil {
+						return nil, err
```

Comment:
I personally really dislike return statements in the middle of functions. There should really be only one return right at the end.
Doing that makes the code easier to follow. Everything should 'drop through' to the final return statement.